src/Expander.hs

```diff
@@ -486,6 +534,15 @@ initializeKernel = do
       , ("Bool", 0, [("true", []), ("false", [])])
       , ("Problem", 0, [("declaration", []), ("expression", [tType]), ("type", []), ("pattern", [])])
       , ("Maybe", 1, [("nothing", []), ("just", [tSchemaVar 0])])
+      , ("List", 1, [("nil", []), ("::", [tSchemaVar 0, Prims.primitiveDatatype "List" [tSchemaVar 0]])])
+      , ("Syntax-Contents"
+        , 1
+        , [ ("list-contents", [Prims.primitiveDatatype "List" [tSchemaVar 0]])
+          , ("integer-contents", [tInteger])
+          , ("string-contents", [tString])
+          , ("identifier-contents", [tString])
```

Comment:
Hmm, I would have expected `identifier-contents` to contain an opaque `Identifier` type holding both an inspectable name and the uninspectable scopes, so that closing a syntax after opening it is the identity. As is, whenever you encounter an identifier you have to be careful to also keep the original `Syntax` so you can re-attach the same scopes when you close it back.

Reply:
We discussed this on a call :-)